apps/admin_app/lib/admin_app_web/views/product_view.ex

```diff
@@ -112,4 +115,74 @@ defmodule AdminAppWeb.ProductView do
     |> Ecto.Query.select([sc], {sc.name, sc.id})
     |> Repo.all()
   end
+
+  def make_search_query_string(conn, column, expression, type, term) do
+    query =
+      "&rummage[search][#{column}][search_expr]=#{expression}&rummage[search][#{column}][search_type]=#{
+        type
+      }&rummage[search][#{column}][search_term]=#{term}"
+
+    conn_query = load_query_string(conn, "rummage", "sort") |> Plug.Conn.Query.encode()
+
+    conn.request_path <> "?" <> conn_query <> query
+  end
+
+  def make_sort_query_string(conn, column, order) do
+    query = "&rummage[sort][field]=#{column}&rummage[sort][order]=#{order}"
+    conn_query = load_query_string(conn, "rummage", "search") |> Plug.Conn.Query.encode()
+
+    conn.request_path <> "?" <> conn_query <> query
+  end
+
+  def load_query_string(conn, parent, key) do
+    Map.new()
```

Comment:
Refactor for easy understanding.
You can use [get_in/2](https://hexdocs.pm/elixir/Kernel.html#get_in/2) and [put_in/2](https://hexdocs.pm/elixir/Kernel.html#put_in/2) for nested map opertions